src/lib/structures/Josh.ts

```diff
@@ -85,7 +85,7 @@ export class Josh<StoredValue = unknown> {
 
     this.middlewares = new MiddlewareStore({ instance: this });
 
-    if (autoEnsure !== undefined) this.use(new AutoEnsure<StoredValue>(autoEnsure));
+    if (autoEnsure !== undefined) this.use(new AutoEnsure(autoEnsure) as unknown as Middleware<NonNullObject, StoredValue>);
```

Comment:
Since it's been moved out of core, this doesn't need the "as unknown as Middleware"

Reply:
Not until after [josh-development/utilities#9](https://github.com/josh-development/utilities/pull/9) is merged. Which is why for now I'm going to mark this PR as blocked.